pytorch_pfn_extras/nn/modules/extended_sequential.py

```diff
@@ -0,0 +1,68 @@
+import torch
+import copy
+from collections.abc import Iterable
+
+
+def _reset_parameters(model):
+    if isinstance(model, torch.nn.Sequential) or \
+       isinstance(model, torch.nn.ModuleList):
+        for submodel in model:
+            _reset_parameters(submodel)
+    elif isinstance(model, torch.nn.ModuleDict):
+        for submodel in model.values():
+            _reset_parameters(submodel)
+    else:
+        if isinstance(model, torch.nn.Module):
+            model.reset_parameters()
+    return model
+
+
+class ExtendedSequential(torch.nn.Sequential):
+    """Sequential module with extended features from chainer.
+
+    """
+    def _copy_model(self, mode):
+        if mode == 'init':
+            return _reset_parameters(copy.deepcopy(self))
+        elif mode == 'copy':
+            return copy.deepcopy(self)
+        else:
```

Comment:
nit:
How about checking the values and raise an error for unknown modes?

Reply:
I think the mode is checked in https://github.com/pfnet/pytorch-pfn-extras/blob/b5bc392feedd0f5c982180d67f11a7d14652f00d/pytorch_pfn_extras/nn/modules/extended_sequential.py#L60

Reply:
Fair enough!